corehq/apps/hqadmin/views/users.py

```diff
@@ -166,12 +174,18 @@ def _update_toggle_edit_permissions(self, selected_tag_slugs, users):
                     permission = ToggleEditPermission(tag_slug=tag.slug)
                 users_to_add = usernames - set(permission.enabled_users)
                 if users_to_add:
+                    for username in users_to_add:
+                        user_changes[username]['added'].append(tag.name)
```

Comment:
is 'added' & 'removed' the standard way?
there is `UserHistory.UPDATE` but not sure that is the right action here.

Reply:
I am mainly interested in how the log_user_change function gets called.

I think the miss here is that the toggle permission is not a "field" on the user, its a related model. So to log changes to it you should look at `UserChangeMessage`, for example https://github.com/dimagi/commcare-hq/blob/873e5200f53be91c25a10f54e4dd79070ec17125/corehq/apps/settings/views.py#L211 for phone number changes